### ISSUE TITLE
message=Error in querying the model., cause=android.database.sqlite.SQLiteException: ambiguous column name: id (code 1 SQLITE_ERROR)

### ISSUE THREAD

**Issue by shirui0413** [aws-amplify/amplify-android#908]
This is my schema  ↓↓↓

```graphql
type User @model {
  id: ID!
  password: String!
  card: [Card] @connection(name: "createdUser")
}

type Card @model {
  id: ID!
  user: User @connection(name: "createdUser")
  date: AWSDateTime!
  category: Int!
  goalItem: Int
  goalValue: Float
  achieve: Float
  reaction: [Reaction] @connection(name: "getReaction")
}

type Reaction @model {
  id: ID!
  card: Card @connection(name: "getReaction")
  category: Int!
  item: Int
}
```

When I used the ID of card as a predicate, there is an error.   ↓↓
![image](https://user-images.githubusercontent.com/27470444/96240619-15ece300-0fdc-11eb-9c2c-a9637d05f125.png)

When I use another field for the predicate, eg. `User` of `Card` as a keyword, there is no problem:
![image](https://user-images.githubusercontent.com/27470444/96241359-015d1a80-0fdd-11eb-9bb8-379603c7e9e9.png)

I don't know why. Who can help me?

**Comment by raphkim** [comment]
Hi @shirui0413 ,

There is currently an issue with the way query works with relational models. Please see the [`**NOTE**` in the following doc](https://docs.amplify.aws/lib/datastore/data-access/q/platform/android#predicates) to work around this issue.

Rather than using `Card.ID` and `Reaction.ID`, please try using `QueryField.field("card.id")` and `QueryField.field("reaction.id")`

**Comment by jamesonwilliams** [comment]
Could we resolve this by changing [this line](https://github.com/aws-amplify/amplify-android/blob/main/aws-datastore/src/main/java/com/amplifyframework/datastore/storage/sqlite/SQLiteCommandFactory.java#L418):
```java
.append(ownedTable.getPrimaryKeyColumnName())
```
to:
```java
.append(ownedTable.getName() + "_" + ownedTable.getPrimaryKeyColumnName())
```
?

**Comment by richardmcclellan** [comment]
The workaround posted above by @raphkim should work, though I am re-opening this issue to track fixing the actual issue, which requires updating the code that generates our Java models, as @jamesonwilliams said.

**Comment by upachecog** [comment]
@richardmcclellan I found an issue in **version 1.6.10.**

When I upgraded the version my queries stopped working:

`Where.matches(QueryField.field("Activity_id").eq(activityId))`
`Where.matches(QueryField.field("Location.id").eq(taskId))`

Neither of those workarounds works in that version.

**Error:**

```DataStoreException{message=Error in saving the model: Activity[id=c0576575-bf0c-4383-be4b-c3141e670608], cause=java.lang.RuntimeException: DataStoreException{message=Error in querying the model., cause=android.database.sqlite.SQLiteException: ambiguous column name: id (code 1 SQLITE_ERROR): , while compiling: SELECT `Activity`.`id` AS `Activity_id`...```

**Comment by raphkim** [comment]
Hi @upachecog ,
Thank you for reporting an error with our latest release. Would you be able to provide the full log to help us identify the root cause? I would specifically need the following:
* The full SQL query that failed to compile (it seems to be cut off in the error message you provided)
* The model details of `Activity` and `Location`

**Comment by richardmcclellan** [comment]
Hi @upachecog, as you noted, there was a regression in 1.6.10, which broke the workaround which you had in place.  This regression has been fixed in the latest version, [1.16.11](https://github.com/aws-amplify/amplify-android/releases/release_v1.16.11), so please give it a try and let us know if it is resolved for you.

Additionally, we are in the process of making some further changes so that the workaround won't be required at all.  Instead, you will be able to do something like this:
```
Activity.ID.eq(activityId)
Location.ID.eq(taskId)
```

**Comment by raphkim** [comment]
> Additionally, we are in the process of making some further changes so that the workaround won't be required at all. Instead, you will be able to do something like this:

The codegen changes are now live with amplify-codegen@2.22.1.

Use the latest version of Amplify CLI (currently on [v4.45.0](https://github.com/aws-amplify/amplify-cli/releases/tag/v4.45.0)) to use enhanced model fields. Note that you have to update your Amplify Android library version to at least v1.16.11 (currently on [v1.17.0](https://github.com/aws-amplify/amplify-android/releases/tag/release_v1.17.0)) to be able to use this feature.

Now instead of `field("Activity_id").eq("123")`, you can directly use the static field from your classes like this: `Activity.ID.eq("123")`.
